enumerable/queue.go

```diff
@@ -0,0 +1,113 @@
+package enumerable
+
+// Queue is an extention of the enumerable interface allowing individual
+// items to be added into the enumerable.
+//
+// Added items will be yielded in a FIFO order.  Items may be added after
+// enumeration has begun.
+type Queue[T any] interface {
+	Enumerable[T]
+	// Put adds an item to the queue.
+	Put(T) error
+	// Size returns the current length of the backing array.
+	//
+	// This may include empty space where yield items previously resided.
+	// Useful for testing and debugging.
+	Size() int
+}
+
+// For now, increasing the size one at a time is likely optimal
+// for the only useage of the queue type.  We may wish to change
+// this at somepoint however.
+const growthRate int = 1
+
+type queue[T any] struct {
+	values       []T
+	currentIndex int
+	lastSetIndex int
```

Comment:
I think read/write is less accurate though, especially given that the two props actually differ in terms of temporality: `currentIndex` *is* the last index produced by a call to `Next`, that may or may not be out of bounds, and may or may not have been read via `Value`.  `lastSetIndex` *has* been set, was valid, and is essentially a record of the past.

Writing all that out though, I may have missed something whilst testing.

- [ ] Test post-loop Next=>Put=>Value - I'm pretty sure it is broken, and that a space between these two integers should always been maintained to prevent the premature overwrite of `Value`